File: backend-java/src/main/java/com/example/pricing/entity/PricingResult.java
package com.example.pricing.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 定价结果实体，对应最终输出的价格建议与收益测算。
 */
@Data
@TableName("pricing_result")
public class PricingResult {
    @TableId(type = IdType.AUTO)
    private Long id;

    @TableField("task_id")
    private Long taskId;

    @TableField("execution_id")
    private String executionId;

    @TableField("final_price")
    private BigDecimal finalPrice;

    @TableField("expected_sales")
    private Integer expectedSales;

    @TableField("expected_profit")
    private BigDecimal expectedProfit;

    @TableField("profit_growth")
    private BigDecimal profitGrowth;

    @TableField("is_pass")
    private Integer isPass;

    @TableField("execute_strategy")
    private String executeStrategy;

    @TableField("result_summary")
    private String resultSummary;

    @TableField("review_required")
    private Integer reviewRequired;

    @TableField("applied_previous_price")
    private BigDecimal appliedPreviousPrice;

    @TableField("applied_at")
    private LocalDateTime appliedAt;

    @TableField("applied_by_user_id")
    private Long appliedByUserId;

    @TableField(exist = false)
    private Long productId;

    @TableField(exist = false)
    private String productTitle;

    @TableField(exist = false)
    private BigDecimal originalPrice;

    @TableField(exist = false)
    private Boolean applied;

    @TableField(value = "created_at", fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(value = "updated_at", fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}

File: backend-python/app/models/pricing_result.py
from datetime import datetime
from decimal import Decimal

from sqlalchemy import BigInteger, DateTime, Integer, String, Text, text
from sqlalchemy.dialects.mysql import DECIMAL
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base


class PricingResult(Base):
    __tablename__ = "pricing_result"

    id: Mapped[int] = mapped_column(BigInteger, primary_key=True, autoincrement=True)
    task_id: Mapped[int] = mapped_column(BigInteger, nullable=False)
    final_price: Mapped[Decimal] = mapped_column(DECIMAL(10, 2), nullable=False)
    expected_sales: Mapped[int | None] = mapped_column(Integer, nullable=True)
    expected_profit: Mapped[Decimal] = mapped_column(DECIMAL(12, 2), nullable=False, default=0)
    profit_growth: Mapped[Decimal] = mapped_column(DECIMAL(12, 2), nullable=False, default=0)
    is_pass: Mapped[int] = mapped_column(Integer, nullable=False, default=0, server_default="0")
    execute_strategy: Mapped[str] = mapped_column(String(50), nullable=False, default="人工审核", server_default="人工审核")
    result_summary: Mapped[str | None] = mapped_column(Text, nullable=True)
    review_required: Mapped[int] = mapped_column(Integer, nullable=False, default=1, server_default="1")
    applied_previous_price: Mapped[Decimal | None] = mapped_column(DECIMAL(10, 2), nullable=True)
    applied_at: Mapped[datetime | None] = mapped_column(DateTime, nullable=True)
    applied_by_user_id: Mapped[int | None] = mapped_column(BigInteger, nullable=True)
    created_at: Mapped[datetime] = mapped_column(DateTime, nullable=False, server_default=text("CURRENT_TIMESTAMP"))
    updated_at: Mapped[datetime] = mapped_column(
        DateTime, nullable=False, server_default=text("CURRENT_TIMESTAMP"), server_onupdate=text("CURRENT_TIMESTAMP")
    )
